### ISSUE TITLE
New Form Builder form:Fatal Error

### ISSUE THREAD

**Issue by johnbleck3** [formtools/core#139]
After the latest updates from the core, now when I try to create a new form_builder form and update the fields showing 20 fields per page (not all fields) I get:
**Fatal error: Call to undefined method FormTools\Fields::updateFieldFilters() in /var/www/forms.buffalonews.com/global/code/Forms.class.php on line 1886**

From the individual edit per field, only the display text will save. Form Field and DB Column error
![1-26-2018 10-19-59 pm](https://user-images.githubusercontent.com/12632568/35468181-374fdaa2-02e7-11e8-890e-c261f27266ba.png)

**Comment by johnbleck3** [comment]
Also, with a form with over 100 fields, when "all fields" are shown, the fields do not update when using the update button at the bottom of the page. This was actually a bug in ft2 - when over 100 fields only the individual field edits would actually save - I had to page the fields to get them to update.

**Comment by benkeen** [comment]
Hey John, could you try the latest core version? https://github.com/formtools/core/releases/tag/3.0.0-alpha-20180127

I think maybe the first error was caused by the code in `master` not being quite ready. I think the core should fix it. 

The second error you posted is very interesting... Could you go to your System Check module -> Env Info page -> phpinfo tab and see if you can see "suhosin" mentioned? Suhosin is a PHP module that's sometimes installed on systems to improve security of PHP. One of its settings is this one: 
https://suhosin.org/stories/configuration.html#suhosin-post-max-vars

That limits the max size of POST requests. Just wondering if that's enabled on your system... that would cause the symptoms you noticed. That form is pretty huge and has a whole slew of data being submitted per row, so maybe you've hit the max...?

**Comment by johnbleck3** [comment]
I will re-load the core and let you know tonight.

the system check Enviro info is only available when I physically change the url - is that intentional? I didn't even realize that was available . . . until now of course. Awesome!
Suhosin Extension: Not Installed
But core php
max_input_vars | 1000
Thank you for pointing me in the right direction. It's actually not worth changing as long as the paging works and I understand why it reacts that way. That's on our end. Thanks.

**Comment by johnbleck3** [comment]
Re-loading the master took care of it. Another issue to check off the list! Thanks.

**Comment by johnbleck3** [comment]
Whoops -- Not completely fixed quite yet:

Changing per field using the popup to modify I get
Error: Invalid JSON:
Fatal error: Cannot redeclare FormTools\placeholders() (previously declared in /var/www/forms.buffalonews.com/global/code/Database.class.php:77) in /var/www/forms.buffalonews.com/global/code/Database.class.php on line 77

When I make the changes in the main fields page the fields get updated.

**Comment by johnbleck3** [comment]
I'm trying to get my finger on when this error comes into play - When editing individual fields (but more than one at a time using "Next"). One field at a time doing "save settings" works, but multiple "next" with changes to settings gives intermittent failures with Fatal error in JSON. If I can get more specific I will in another comment. Let me know if you can reproduce.

**Comment by johnbleck3** [comment]
![1-29-2018 6-58-34 pm](https://user-images.githubusercontent.com/12632568/35540968-80a31de0-0526-11e8-91d9-6daf9a05fa37.png)

**Comment by benkeen** [comment]
Huh, curious one... I'll tinker around with this myself and see if I can get it to break.

**Comment by benkeen** [comment]
Hey @johnbleck3 sorry I haven't looked at this one yet - definitely looks like a bug. I'm just 90% through getting the upgrade script back online and want to focus on getting that out the door this weekend. Will return to this soon.

**Comment by johnbleck3** [comment]
no worries! Meanwhile I just make changes to one field at a time. I'll avoid the "Next" link until you get this squared away.

**Comment by benkeen** [comment]
Hey @johnbleck3, I just can't seem to reproduce this problem unfortunately. It's all working perfectly on my end. 

I don't suppose I could get access to your server to debug the issue directly? If so, email me at ben.keen@gmail.com.

**Comment by johnbleck3** [comment]
The fact it's not happening for you is telling. Sounds like it's my environment - there may have been a file transfer error along the way. Let me take another look first and we'll go from there.

**Comment by benkeen** [comment]
Sounds good. Good luck!

**Comment by johnbleck3** [comment]
I haven't found the issue yet, but I have found the following to be true: when editing fields inner_tab1 and inner_tab3 do NOT cause the redeclaration of placeholders() issue. Only inner_tab2 when more than one field is edited/changed. Field 1 change comments. Next: field 2 change comments. Save. Same for any of the other options in inner_tab2. inner_tab1 and inner_tab3 don't throw an error at all when changing multiple fields.

**Comment by johnbleck3** [comment]
My last statement is not entirely true: when editing more than 5 or 6 fields (about 1 to 2 settings per field) I get the error from inner_tab1 and from inner_tab3. inner_tab2 will throw that error when editing just 2 fields at one time. Try to change 20 fields using the edit and Next changing one setting per tab per field. Let me know if it throws the error. If not, let me know if you have any suggestions.

**Comment by benkeen** [comment]
Apologies, @johnbleck3 - you're quite right this is a bug, I was just able to reproduce it. Let me see what I can find.

**Comment by benkeen** [comment]
Ahhh, now I understand. The `placeholders()` msg in the error was due to a function being referenced internally by PHP as a function-in-function. I've refactored the method and all seems good now. 

I just want to do a little more testing but I'll release a fresh Beta including this fix tomorrow.  Very sorry for the long wait on this issue!

**Comment by johnbleck3** [comment]
I'm just thrilled you were able to recreate it and found the root cause.
Awesome - looking forward to having that fixed up - ty!

**Comment by benkeen** [comment]
This is included in today's beta. Let me know if it doesn't fix it for you. Thanks!

**Comment by johnbleck3** [comment]
Looking good! Looks like that did it. - the visualizations got funky on me though. see the comment I just posted in visualizations. Thank you!!
